src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -19,68 +19,104 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Database;
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
```

Comment:
Is this really an integration test ? I imagined this to bring up a local grpc server and test via the public GRPC methods rather than building the spanner daos and testing those methods.

Reply:
This is a little unclear to me as well. It's not quite a unit test since it's running through both SpannerDao and the emulator. In my chats recently with @voulg and @c24t, we've been referring to this as an integration test, although it doesn't integrate GRPC but still tests the emulator and our code working together. So I believe it still merits this label because it's integrating the emulator and our code.

A full end-to-end test is ideal for a final product, but for this PR at least, we'll stick to these SpannerDao tests.

Reply:
Using com.google.cloud.spanner.IntegrationTestEnv should make this test able to run with either emulator or spanner. Moreover emulator (or real spanner) is considered part of the environment and is initialized outside the test (e.g: https://github.com/googleapis/java-spanner-jdbc/blob/master/.github/workflows/integration-tests-against-emulator.yaml), so IT is appropriate.

If the test was bringing emulator up internally I'd agree it's not a very clear cut.

Reply:
Acknowledged.